src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/ClientCreds/JsonWebToken.cs

```diff
@@ -129,6 +131,12 @@ private string Encode(IClientAssertionCertificate credential)
             // Header segment
             string jsonHeader = EncodeHeaderToJson(credential);
 
+            //Remove x5c claim if no certificate is available
+            if (jsonHeader.Contains("no_cert"))
+            {
+                string emptyX5cClaim = "\"x5c\":\"no_cert\",";
+                jsonHeader = jsonHeader.Remove(jsonHeader.IndexOf(emptyX5cClaim), emptyX5cClaim.Length);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define the locale to be used in this string operation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS1449)